src/frameworks/foundation/ns_timer.rs

```diff
@@ -118,6 +118,12 @@ pub const CLASSES: ClassExports = objc_classes! {
     env.objc.borrow::<NSTimerHostObject>(this).due_by.is_some()
 }
 
+-(())invalidate {
+    // Remove the timer from the run loop.
+    let run_loop: id = msg_class![env; NSRunLoop currentRunLoop];
+    ns_run_loop::remove_timer(env, run_loop, this);
```

Comment:
I think you also need to ensure `isValid` above will return `false` in this case.